pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -325,6 +323,10 @@ def _apply_lightning(
                 operation.base, qml.GlobalPhase
             ):  # apply n-controlled gate
                 # LGPU do not support the controlled gates except for GlobalPhase
+                if self._mpi_handler.use_mpi:
+                    raise DeviceError(
+                        "Lightning-GPU-MPI does not support Controlled GlobalPhase gates."
+                    )
```

Comment:
The codefactor error in line 281 can probably be be circumvented if you merge lines 312 and 313.